integration-tests/src/integrationTest/java/pl/allegro/tech/hermes/integrationtests/management/TopicManagementTest.java

```diff
@@ -0,0 +1,624 @@
+        Topic untrackedJsonTopic = hermes.initHelper().createTopic(topicWithRandomName().withTrackingEnabled(false).withContentType(JSON).build());
+        Topic trackedJsonTopic = hermes.initHelper().createTopic(topicWithRandomName().withTrackingEnabled(true).withContentType(JSON).build());
+
+        // and
+        String query = "{\"query\": {\"trackingEnabled\": \"true\", \"contentType\": \"AVRO\"}}";
+
+        // when
+        WebTestClient.ResponseSpec response = hermes.api().queryTopics("", query);
+
+        // then
+        assertThat(Arrays.stream(Objects.requireNonNull(response.expectBody(String[].class).returnResult().getResponseBody())).toList())
+                .contains(trackedTopicAvro.getQualifiedName())
+                .doesNotContain(untrackedJsonTopic.getQualifiedName(), untrackedAvroTopic.getQualifiedName(), trackedJsonTopic.getQualifiedName());
+    }
+
+    @Test
+    public void shouldReturnTrackedTopicsWithAvroContentTypeForGivenGroup() {
+        // given
+        String group = "mixedTrackedGroup3";
+        Topic trackedTopicAvro = hermes.initHelper().createTopicWithSchema(new TopicWithSchema(topic(group, "trackedAvroTopic3").withTrackingEnabled(true).withContentType(AVRO).build(), SCHEMA));
+        hermes.api().createTopic(new TopicWithSchema(topic(group, "untrackedAvroTopic").withTrackingEnabled(false).withContentType(AVRO).build(), SCHEMA));
+        hermes.api().createTopic(new TopicWithSchema(topic(group, "untrackedJsonTopic").withTrackingEnabled(false).withContentType(JSON).build(), null));
+        hermes.api().createTopic(new TopicWithSchema(topic(group, "trackedJsonTopic").withTrackingEnabled(true).withContentType(JSON).build(), null));
+
+        // and
+        String query = "{\"query\": {\"trackingEnabled\": \"true\", \"contentType\": \"AVRO\"}}";
+
+        // when
+        WebTestClient.ResponseSpec response = hermes.api().queryTopics(group, query);
+
+        // then
+        assertThat(Arrays.stream(Objects.requireNonNull(response.expectBody(String[].class).returnResult().getResponseBody())).toList())
+                .containsExactly(trackedTopicAvro.getQualifiedName())
+                .doesNotContain(group + "." + "untrackedAvroTopic", group + "." + "untrackedJsonTopic", group + "." + "trackedJsonTopic");
+    }
+
+    @Test
+    public void shouldNotAllowDollarSign() {
+        // given
+        String group = "dollar";
+        hermes.initHelper().createGroup(Group.from(group));
+
+        Stream.of("$name", "na$me", "name$").forEach(topicName -> {
+            // when
+            WebTestClient.ResponseSpec response = hermes.api().createTopic(new TopicWithSchema(topic(group, topicName).build(), null));
+
+            // then
+            response.expectStatus().isBadRequest();
+        });
+    }
+
+//    @Test
+//    public void shouldCreateTopicEvenIfExistsInBrokers() {
```

Comment:
what is the reason for commenting them out?

Reply:
I haven't rewritten them for the new solution yet because we don't have a framework developed for broker operations. Good point, I removed the commented tests